asmcli/lib/checks.sh

```diff
@@ -464,12 +464,15 @@ istio_namespace_exists() {
 }
 
 is_autopilot() {
-  # TODO: wait till https://pkg.go.dev/google.golang.org/genproto/googleapis/container/v1#Cluster  
-  # publish the `Autopilot` field
-  # This is a temporary workaround to check if a cluster is Autopilot or GKE
-  # This CRD will be installed only if the cluster is Autopilot
   if [[ "${IS_AUTOPILOT}" -eq 1 ]]; then return; fi
-  if ! retry 2 kubectl get crd allowlistedworkloads.auto.gke.io 1>/dev/null 2>/dev/null; then
+
+  local AUTOPILOT_MODE; AUTOPILOT_MODE="$(gcloud container clusters describe "${CLUSTER_NAME}" \
+    --region "${CLUSTER_LOCATION}" \
+    --project "${PROJECT_ID}" \
+    --format json | \
+    jq '.autopilot.enabled')"
+
+  if [[ "${AUTOPILOT_MODE}" = 'null' ]]; then
```

Comment:
will `${AUTOPILOT_MODE}` ever be `false`? maybe we should invert the check to be  `!= 'true'`.

Reply:
+1. I can approve/merge after this change.

Reply:
thanks, updated